sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1563,13 +1563,29 @@ class Analyzer(override val catalogManager: CatalogManager)
                 }
                 InsertAction(
                   resolvedInsertCondition,
-                  resolveAssignments(assignments, m, resolveValuesWithSourceOnly = true))
+                  resolveAssignments(assignments, m, resolveValuesFrom = sourceTable))
               case o => o
             }
+            val newNotMatchedBySourceActions = m.notMatchedBySourceActions.map {
+              case DeleteAction(deleteCondition) =>
+                val resolvedDeleteCondition = deleteCondition.map(
+                  resolveExpressionByPlanChildren(_, Project(Nil, targetTable)))
```

Comment:
nit: we can call `resolveExpressionByPlanOutput`, then we don't need to wrap the plan with `Project(Nil, ...)`

Reply:
Updated.